packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -539,11 +539,10 @@ export class ProfilesCache {
     }
 
     private shouldRemoveTokenFromProfile(profile: zowe.imperative.IProfileLoaded, baseProfile: zowe.imperative.IProfileLoaded): boolean {
-        return (baseProfile?.profile?.host &&
-            baseProfile?.profile?.port &&
+        return ((baseProfile?.profile?.host || baseProfile?.profile?.port) &&
             profile?.profile?.host &&
             profile?.profile?.port &&
-            (baseProfile?.profile.host !== profile?.profile.host || baseProfile?.profile.port !== profile?.profile.port) &&
```

Comment:
Can we add this check back for comparing the base profile's port with the port of the profile that was passed in? I like the added check for user and password on the profile, but I think keeping the port check will help narrow the opportunity for error until something more comprehensive is incorporated into Zowe CLI/Imperative.